File: backend/templates/backend/node/18/crud-service/src/services/itemService.ts
import { Item, IItem } from '../models/Item.js';

export class ItemService {
  async getAll(): Promise<IItem[]> {
    return Item.find().sort({ createdAt: -1 });
  }

  async getById(id: string): Promise<IItem | null> {
    return Item.findById(id);
  }

  async create(data: Partial<IItem>): Promise<IItem> {
    const item = new Item(data);
    return item.save();
  }

  async update(id: string, data: Partial<IItem>): Promise<IItem | null> {
    return Item.findByIdAndUpdate(id, data, { new: true, runValidators: true });
  }

  async delete(id: string): Promise<boolean> {
    const result = await Item.findByIdAndDelete(id);
    return !!result;
  }
}

File: backend/templates/backend/springboot/2.7/crud-service/src/main/java/com/example/crud/service/ItemService.java
package com.example.crud.service;

import com.example.crud.model.Item;
import com.example.crud.repository.ItemRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ItemService {

    @Autowired
    private ItemRepository itemRepository;

    public List<Item> getAll() {
        return itemRepository.findAll();
    }

    public Optional<Item> getById(Long id) {
        return itemRepository.findById(id);
    }

    public Item create(Item item) {
        return itemRepository.save(item);
    }

    public Optional<Item> update(Long id, Item itemDetails) {
        return itemRepository.findById(id).map(item -> {
            item.setName(itemDetails.getName());
            item.setDescription(itemDetails.getDescription());
            item.setPrice(itemDetails.getPrice());
            return itemRepository.save(item);
        });
    }

    public boolean delete(Long id) {
        if (itemRepository.existsById(id)) {
            itemRepository.deleteById(id);
            return true;
        }
        return false;
    }
}

